### ISSUE TITLE
Call to Node module failed with error: TypeError: Converting circular structure to JSON

### ISSUE THREAD

**Issue by Maarten88** [aspnet/JavaScriptServices#987]
I am trying to implement webpack code splitting in a React-Redux based Javascriptservices project.

So I created an asyncComponent using one of the many samples on the web, and made a simple Contact page load dynamically, like this:

    const AsyncContact = asyncComponent(() => 
        System.import('./components/Contact').then(module => module.default));

This works, both client-side and serverside.

For a more complex page, with a lot of dependencies, I implemented the same, now with dynamic reducers. and a large Chat component on it. Clientside this page runs without error, but it results in an error when run serverside:

    Call to Node module failed with error: TypeError: Converting circular structure to JSON

This error seems to come from here:
https://github.com/aspnet/JavaScriptServices/blob/dev/src/Microsoft.AspNetCore.NodeServices/Content/Node/entrypoint-http.js#L78

If I refresh the page, the second time the same page renders succesfully.

I guess my webpack chunk has circular references. I tried to find out which, but I don't know how to debug this. I think Javascript should normally handle circular references, the browser has no problems with it.

**Comment by Maarten88** [comment]
I was finally able to debug the issue using node-inspector; it turned out the state picked up a datastructure that was meant to be created only client side, but it ended up server-side.

A better error message would have saved me time, I never understood what data was being serialized there.

Closing the issue.

**Comment by timothywisdom** [comment]
@Maarten88 - I'm curious about your solution as I have a similar problem. I've reduced it down to a server-side fetch operation but I think this entrypoint-http.js file is accessing some other data structure as my fetch is fairly moot.

**Comment by Maarten88** [comment]
@timothywisdom, well, start your project with NodeServices in debug configuration, start chrome, connect the Chrome debugger to node (a green node icon should appear upper-left in the F12 tools menu), find your entrypoint in the transpiled javascript and set a breakpoint there to see what is going wrong. That's how I found my error.
In my case, it was a bug that I made, the bug had nothing to do with the error message.
